extensions/ql-vscode/src/view/results/__tests__/results.spec.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from "react";
+import { render as reactRender, screen } from "@testing-library/react";
+import { ResultsApp } from "../results";
+import {
+  Interpretation,
+  IntoResultsViewMsg,
+  SortDirection,
+} from "../../../pure/interface-types";
+import * as fs from "fs-extra";
+import { resolve } from "path";
+import { ColumnKindCode } from "../../../pure/bqrs-cli-types";
+
+const exampleSarif = fs.readJSONSync(
+  resolve(
+    __dirname,
+    "../../../../test/vscode-tests/no-workspace/data/sarif/validSarif.sarif",
+  ),
+);
+
+describe(ResultsApp.name, () => {
+  const render = () => reactRender(<ResultsApp />);
+  const postMessage = async (msg: IntoResultsViewMsg) => {
+    // window.postMessage doesn't set the origin correctly, see
+    // https://github.com/jsdom/jsdom/issues/2745
+    window.dispatchEvent(
+      new MessageEvent("message", {
+        source: window,
+        origin: window.location.origin,
+        data: msg,
+      }),
+    );
+
+    // The event is dispatched asynchronously, so we need to wait for it to be handled.
+    await new Promise((resolve) => setTimeout(resolve, 0));
```

Comment:
This seems brittle to me. Are we guaranteed that the dispatch will complete before the next tick?

If this is indeed running in a DOM context, does it make sense to add an event handler that resolves the promise? If it seems to work reliably right now, then I'm fine to keep as it is, but this may be something we need to revisit.